src/object.js

```diff
@@ -16,6 +16,50 @@ Sk.builtin.object = function () {
     return this;
 };
 
+Object.defineProperties(Sk.builtin.object.prototype, /**@lends {Sk.builtin.object.prototype}*/ {
+    ob$type: { value: Sk.builtin.object, writable: true },
+    tp$name: { value: "object", writable: true },
+    tp$base: { value: undefined, writable: true },
+    sk$object: { value: true },
+});
+
+/**
+ * @description
+ * We aim to match python and javascript inheritance like
+ * type   instanceof object => true
+ * object instanceof type   => true
+ * type   instanceof type   => true
+ * object instanceof object => true
+ *
+ * type   subclassof object => type.prototype   instanceof object => true
+ * object subclassof type   => object.prototype instanceof type   => false
+ * 
+ * this algorithm achieves the equivalent with the following prototypical chains
+ * using `Object.setPrototypeOf`
+ *
+ * ```
+ * type.__proto__             = type   (type instanceof type)
+ * type.__proto__.__proto__   = object (type instanceof object)
+ * type.prototype.__proto__   = object (type subclasssof object)
+ * object.__proto__           = type   (object instanceof type)
+ * object.__proto__.__proto__ = object (object instanceof object)
+ * ```
+ *
+ * while `Object.setPrototypeOf` is not considered [good practice](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/setPrototypeOf)
+ * this is a particularly unique use case and creates a lot of prototypical benefits
+ * all single inheritance classes (i.e. all builtins) now follow prototypical inheritance
+ * similarly it makes metclasses that much easier to implement
+ * Object.setPrototypeOf is also a feature built into the javascript language
+ *
+ * @ignore
+ */
+(function setUpBaseInheritance () {
+    Object.setPrototypeOf(Sk.builtin.type.prototype, Sk.builtin.object.prototype);
+    Object.setPrototypeOf(Sk.builtin.type, Sk.builtin.type.prototype);
+    Object.setPrototypeOf(Sk.builtin.object, Sk.builtin.type.prototype);
+    Sk.builtin.type.prototype.tp$base = Sk.builtin.object;
+})();
+
```

Comment:
here's where the magic happens

Reply:
No kidding. :exploding_head: 

Comment needs updating to (I think):
```
 * type.__proto__             = type.prototype   (type instanceof type)
 * type.__proto__.__proto__   = object.prototype (type instanceof object)
 * type.prototype.__proto__   = object.prototype (type subclassof object)
 * object.__proto__           = type.prototype   (object instanceof type)
 * object.__proto__.__proto__ = object.prototype (object instanceof object)
```